akka-docs/rst/scala/stream/stream-customize.rst

```diff
@@ -385,7 +385,7 @@ or ``unwatch(ref)`` methods. The reference can be also watched by external actor
 Custom materialized values
 --------------------------
 
-Custom stages can return materialized values instead of ``Unit`` by inheriting from :class:`GraphStageWithMaterializedValue`
+Custom stages can return materialized values instead of ``NotUsed`` by inheriting from :class:`GraphStageWithMaterializedValue`
```

Comment:
[This](http://doc.akka.io/docs/akka/2.4/scala/stream/stream-customize.html#Custom_materialized_values) refers to [`GraphStageDocTest.java`](https://github.com/akka/akka/blob/4a7cd147313404eb8137e56163e54f06c23640d0/akka-docs/rst/java/code/docs/stream/GraphStageDocTest.java#L577)

```
class FirstValue[A] extends GraphStageWithMaterializedValue[FlowShape[A, A], Future[A]] {
```

As in the previous comment,  [`GraphStage`](https://github.com/akka/akka/blob/4a7cd147313404eb8137e56163e54f06c23640d0/akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala#L42)is defined as a specialization of `GraphStageWithMaterializedValue` which materializes to `NotUsed`:

```
abstract class GraphStage[S <: Shape] extends GraphStageWithMaterializedValue[S, NotUsed] {
```

So, inheriting from `GraphStageWithMaterializedValue` allows your stage materializes to a different type.